app/models/comment.rb

```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :users, foreign_key: :author_id, inverse_of: :comments
+  belongs_to :posts, inverse_of: :comments
+
+  def update_comments_count
+    post = Post.find(post_id)
+    comments_count = Comment.where(post_id: post_id).count
+    post.update(comments_counter: comments_count)
+  end
```

Comment:
- [OPTIONAL] Here, we can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private. Your model would end up looking like this:

```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```